client/src/actions/authActions.js

```diff
@@ -107,39 +123,61 @@ const logout = () => (dispatch) => {
 
 /**
  * Registers User
+ *
  * @description Gets user details by registering them
+ *
  * @param {object} userData - user details
+ *
  * @returns {object} action
  */
 const userSignUpRequest = userData =>
-  axios.post('/api/v1/users/signup', userData)
-    .then((res) => {
-      Materialize.toast(`${res.data.message} You're logged in`, 3000, 'green');
-      return {
-        isDone: res.data.success,
-        message: res.data.message
-      };
-    })
-    .catch(err => Materialize.toast(err.response.data.message, 3000, 'red'));
+  dispatch =>
+    axios.post('/api/v1/users/signup', userData)
+      .then((res) => {
+        const { message } = res.data;
+        const { username, password } = userData;
+        const loginDetails = { identifier: username, password };
+        return dispatch(login(loginDetails)).then((res) => {
+          Materialize.toast(`${message} You're logged in`, 3000, 'green');
+          return { isDone: res.isAuthenticated, message };
+        });
+      })
+      .catch((err) => {
+        Materialize.toast(err.response.data.message, 3000, 'red');
+        return err.response.data;
+      });
 
 
 /**
  * Check User existence
+ *
  * @description Checks if a users exists in the library
+ *
  * @param {object} userData - user details
+ *
  * @returns {object} action
  */
 const isUserExists = userData =>
```

Comment:
i think you might want to find a better name for this function as a function mostly depletes performing an action

Reply:
thanks just did that now changed it to `doesUserExist`